src/main/java/com/smartbear/readyapi/plugin/git/GitCommandHelper.java

```diff
@@ -375,4 +380,57 @@ private boolean canMerge(final Git git, final MergeStrategy mergeStrategy) {
 
         return true;
     }
+
+    public String getCurrentBranch(final Git git) {
+        try {
+            return git.getRepository().getFullBranch();
+        } catch (IOException e) {
+            throw new VcsIntegrationException("Can get current branch", e);
+        }
+    }
+
+    public List<String> getBranchList(final Git git) {
+        try {
+            return git.branchList()
+                    .setListMode(ListBranchCommand.ListMode.ALL)
+                    .call()
+                    .stream()
+                    .map(Ref::getName)
+                    .collect(Collectors.toList());
+        } catch (GitAPIException e) {
+            throw new VcsIntegrationException("Can not get list of branches", e);
+        }
+    }
+
+    public void checkout(String commitOrBrunch, final Git git) {
+        if (commitOrBrunch == null) {
+            throw new NullPointerException("commitOrBrunch is null");
```

Comment:
I think, the IllegalArgumentException is more applicable in this case.

Reply:
Replaced